### ISSUE TITLE
Conversion of "Save uncombined" series not working as expected

### ISSUE THREAD

**Issue by mharms** [rordenlab/dcm2niix#235]
I have some single slice data collected on a phantom on a Siemens VE11C with a non-Siemens 32 ch head coil, for which I used the "Save uncombined" sequence option to look at the signal profiles of the individual channels.  So, I was expecting 32 separate NIFTIs, but I'm getting only 12 using a compiled version of the current master (version v1.0.20180931)

The messages that I'm getting are:
```
slices not stacked: coil varies
1 images have identical time, series, acquisition and instance values. DUPLICATES REMOVED.
```
Would it be helpful for me to load the DICOMs somewhere? (They are only 2.8 MB).

Also, we might want to revisit the setting of `ReceiveCoilActiveElements`.  Back in #127 I had suggested:

> ReceiveCoilActiveElements: sCoilSelectMeas.sCoilStringForConversion OR (0051,100f)

Those two are equal when NOT using "Save uncombined".  But for a series representing the uncombined channel images, I now see that only (0051,100f) contains a representation of the name of the actual individual channel (which would be appropriate IMO to use as the  `ReceiveCoilActiveElements` entry corresponding to json/NIFTI for individual channels).

**Comment by neurolabusc** [comment]
In the past, I detected uncombined data by looking for a unique number in the coil name (0051,100F), [though see 187](https://github.com/rordenlab/dcm2niix/issues/187). The danger of relying on private tags is that the vendor can set them to whatever they want. In your case, the 32 coil elements of the coil `PHC` do not have unique numbers, and in fact some one of the coils does not have a number in its name (`UC`).  Here are your 32 coil names: `[C1, C2, C3, L01, L02, L03, L04, L05, L06, L07, L08, L09, L10, L11, R01, R02, R03, R04, R05, R06, R07, R08, R09, R10, R11, UC, UL1, UL2, UL3, UR1, UR2, UR3]`. 

To deal with this, I now use the CRC32 of the coil name to detect if the coil varies, and if it does the coil name (rather than number) is appended to the file name. I have also updated the BIDS file - as you note, the ReceiveCoilActiveElements name found in the ASCII portion of the CSA header does not necessarily match the  coil name (0051,100F). The latest BIDS output now includes both names, for example
 
```
	"ReceiveCoilActiveElements": "PHC",
	"CoilString": "C3",
```

In general, I think we are ready for a new stable release, and this was a big change to the code at such a late stage. I would be grateful if you could vet these changes carefully.

**Comment by mharms** [comment]
Behaves appropriately and reasonably on my couple test series.  As usual, thanks for the very quick response!!

**Comment by neurolabusc** [comment]
@mharms The default file naming system has not changed, it is '%f_%p_%t_%s'. Is it possible that you generated a new set of default naming scheme by using `-g y`? The upcoming version has the following changes:
- Using `%a` in for the file name now stores the antenna name (e.g. `L02`), not number (`2`) to disambiguate the situations you described above.
- If `%a` if not explicitly included in the filename but antenna name varies within a series the antenna name will be appended at the end of the filename. This disambiguates uncombined data.
- If slices are not parallel and `%r` (instance number) if not explicitly included in the filename the instance number will be appended at the end of the filename. This disambiguates orthogonal slices of a localizer scan that can not be stacked in the NIfTI format (as NIfTI requires all slices are co-planar). Personally, I always ignore localizers (`-i y`) as they are typically only used for planning and not analyses.

These are both cases that are not typically used in pipelines: only advanced users tend to acquire uncombined data and virtually everyone discards localizers. I did ask members of some of the large, sophisticated studies to comment on these changes. I appreciate your concern

Well spotted, `ReceiveCoilActiveElements` has been re-introduced. I have also included the new tag `CoilString`. With the current code, `CoilString` is only added if it is different from `ReceiveCoilName`. Feel free to comment on this choice: should we show both tags when they are redundant or only show both if they provide unique information. As always, I let @chrisfilo make the final decision on any new addition, even if it is not part of the BIDS standard. 

I have also added the unofficial tag `ImagingFrequency` based on the DICOM tag of the same name (0018,0084) - this tends to provide more precision than the `MagneticFieldStrength` which tends to be rounded up.

```
	"MagneticFieldStrength": 3,
	"ImagingFrequency": 123.248,
```

**Comment by mharms** [comment]
Are you sure about that?  I just compiled v1.0.20181006 and according to its own usage statement

>   -f : filename (%a=antenna  (coil) name, %b=basename, %c=comments, %d=description, %e=echo number, %f=folder name, %i=ID of patient, %j=seriesInstanceUID, %k=studyInstanceUID, %m=manufacturer, %n=name of patient, %p=protocol, %r=instance number, %s=series number, %t=time, %u=acquisition number, %v=vendor, %x=study ID; %z=sequence name; **default '%p_%s_%a'**)

For good measure, I ran '-g i':

> Chris Rorden's dcm2niiX version v1.0.20181006  GCC4.4.7 (64-bit Linux)
> Defaults reset
> Example output filename: '/myFolder_MPRAGE_19770703150928_1_c.nii'

and then ran a conversion with no -f flag on a series that did NOT involve individual coil channels, and I got outputs named as '%p_%s_%a' -- i.e., exactly like the usage statement for the -f flag currently says to expect for the default naming (and not '%f_%p_%t_%s', which is indeed what the output of '-g i' suggests that it should have been).  So, there appears to be a disconnect in the code related to the default naming at this particular point in time.

As for `CoilString`, I don't really care, but it seems reasonable to me to only include it in the situation when it is different from `ReceiveCoilActiveElements` (not `ReceiveCoilName`, which is a typo above). Based on what we currently know, I believe this means that `CoilString` will only appear in the situation when you have individual channel data.
